File: 1403. Minimum Subsequence in Non-Increasing Order.kt
class Solution {
    fun minSubsequence(nums: IntArray): List<Int> {
    val res = mutableListOf<Int>()
    nums.sortDescending()
    var sum = nums.sum()
    var cur = 0
    for (i in nums.indices) {
        cur += nums[i]
        res.add(nums[i])
        if (cur > sum - cur) return res
    }
    return res
    }
}

File: 1403. Minimum Subsequence in Non-Increasing Order.rs
impl Solution {
    pub fn min_subsequence(nums: Vec<i32>) -> Vec<i32> {
    let mut res: Vec<i32> = Vec::new();
    let mut nums = nums;
    nums.sort_by(|a, b| b.cmp(a));
    let sum: i32 = nums.iter().sum();
    let mut cur = 0;
    for i in 0..nums.len() {
        cur += nums[i];
        res.push(nums[i]);
        if cur > sum - cur {
            return res;
        }
    }
    res 
    }
}
